Sources/StytchUI/StytchB2BUIClient/ViewControllers/SMSOTPEntryViewController/SMSOTPEntryViewController.swift

```diff
@@ -17,20 +21,78 @@ final class SMSOTPEntryViewController: BaseViewController<SMSOTPEntryState, SMSO
         stackView.spacing = .spacingRegular
 
         stackView.addArrangedSubview(titleLabel)
+
+        let smsConfirmationLabel = UILabel.makeComboLabel(
+            withPlainText: "A 6-digit passcode was sent to you at",
+            boldText: MemberManager.phoneNumber,
+            fontSize: 18,
+            alignment: .left
+        )
+        stackView.addArrangedSubview(smsConfirmationLabel)
+
         let otpView = OTPCodeEntryView(frame: .zero)
         otpView.delegate = self
         stackView.addArrangedSubview(otpView)
 
+        stackView.addArrangedSubview(expiryButton)
+
         stackView.addArrangedSubview(SpacerView())
 
         attachStackView(within: view)
 
+        updateExpiryText()
+        timer = Timer.scheduledTimer(timeInterval: 1, target: self, selector: #selector(updateExpiryText), userInfo: nil, repeats: true)
+
+        NSLayoutConstraint.activate([
+            otpView.heightAnchor.constraint(equalToConstant: 50),
+        ])
+
         NSLayoutConstraint.activate(
             stackView.arrangedSubviews.map { $0.widthAnchor.constraint(equalTo: stackView.widthAnchor) }
         )
     }
 }
 
+extension SMSOTPEntryViewController: OTPEntryViewControllerProtocol {
+    var expirationDate: Date {
+        viewModel.state.expirationDate
+    }
+
+    func resendCode() {
+        Task {
+            do {
+                if let phoneNumberE164 = MemberManager.phoneNumber {
+                    try await AuthenticationOperations.smsSend(phoneNumberE164: phoneNumberE164)
+                }
+            } catch {
+                presentErrorAlert(error: error)
+            }
+        }
+    }
+
+    func presentCodeResetConfirmation() {
+        guard let phoneNumber = MemberManager.phoneNumber else {
+            return
+        }
+
+        presentCodeResetConfirmation(message: .localizedStringWithFormat(
+            NSLocalizedString("stytch.otpNewCodeWillBeSent", value: "A new code will be sent to %@.", comment: ""), phoneNumber
+        ))
+    }
+
+    @objc private func updateExpiryText() {
+        updateExpirationText()
+    }
+}
+
 extension SMSOTPEntryViewController: OTPCodeEntryViewDelegate {
-    func didEnterOTPCode(_: String) {}
+    func didEnterOTPCode(_ code: String) {
```

Comment:
This `_` shouldn't be here, should it?

Reply:
yes this is wrong, fixing now, good catch!

Reply:
No this is fine, what that means is when you call the function you don't need the parameter label so in this case its `didEnterOTPCode(code)` if the function definition was `func didEnterOTPCode(code: String)` then it would be `didEnterOTPCode(code: code)`